iphone/Classes/TiUIPicker.m

```diff
@@ -26,55 +26,55 @@ -(void)dealloc
 
 USE_PROXY_FOR_VERIFY_AUTORESIZING
 
--(CGFloat)verifyHeight:(CGFloat)suggestedHeight
-{
-	// pickers have a forced height so we use it's height
-	// instead of letting the user set it
-	return picker.frame.size.height;
-}
-
--(CGFloat)verifyWidth:(CGFloat)suggestedWidth
-{
-    if (suggestedWidth <= 0 && picker != nil) {
-        return picker.frame.size.width;
-    }
-    else {
-        return suggestedWidth;
-    }
-}
-
 -(UIControl*)picker 
 {
 	if (picker==nil)
 	{
-        float width = 320;
-        float height = 216;
         
-        if ([TiUtils isIOS9OrGreater]) {
-            width = [TiUtils floatValue:[self.proxy valueForKey:@"width"] def:320];
-            height = [TiUtils floatValue:[self.proxy valueForKey:@"height"] def:216];
-        }
-    
-        [[self proxy ]setValue:NUMDOUBLE(width) forKey:@"width"];
-        [[self proxy ]setValue:NUMDOUBLE(height) forKey:@"height"];
-            
         if (type == -1)
 		{
 			//TODO: this is not the way to abstract pickers, note the cast I had to add to the following line
-			picker = (UIControl*)[[UIPickerView alloc] initWithFrame:CGRectMake(0, 0, width, height)];
+			picker = (UIControl*)[[UIPickerView alloc] initWithFrame:[self bounds]];
 			((UIPickerView*)picker).delegate = self;
 			((UIPickerView*)picker).dataSource = self;
 		}
 		else 
 		{
-			picker = [[UIDatePicker alloc] initWithFrame:CGRectMake(0, 0, width, height)];
+			picker = [[UIDatePicker alloc] initWithFrame:[self bounds]];
 			[(UIDatePicker*)picker setTimeZone:[NSTimeZone localTimeZone]];
 			[(UIDatePicker*)picker setDatePickerMode:type];
 			[picker addTarget:self action:@selector(valueChanged:) forControlEvents:UIControlEventValueChanged];
 		}
 		[picker setBackgroundColor:[UIColor whiteColor]];
 		[self addSubview:picker];
-	}
+#ifndef TI_USE_AUTOLAYOUT
+        [self setTranslatesAutoresizingMaskIntoConstraints:NO];
+        [self addConstraints: [NSLayoutConstraint constraintsWithVisualFormat:@"V:|[picker]|" options:NSLayoutFormatAlignAllLeft metrics:0 views:NSDictionaryOfVariableBindings(picker)]];
+        [self addConstraints: [NSLayoutConstraint constraintsWithVisualFormat:@"H:|[picker]|" options:NSLayoutFormatAlignAllLeft metrics:0 views:NSDictionaryOfVariableBindings(picker)]];
+        
+        [self setNeedsLayout];
+        [self layoutIfNeeded];
+        [self layoutSubviews];
+        float width  =  self.bounds.size.height;
+        float height = self.bounds.size.width;
+        if (![TiUtils isIOS9OrGreater])
+        {
+            width  = 320;
+            height = 216;
+            //TIMOB-20028
+            [[self proxy ]setValue:NUMDOUBLE(width) forKey:@"width"];
+            [[self proxy ]setValue:NUMDOUBLE(height) forKey:@"height"];
+        }
```

Comment:
As discussed in 1:1, we should migrate the Picker to use the layout engine instead of handling all dimensions manually. For example, what do we do if the user sets `Ti.UI.FILL` on the picker?